lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -865,3 +865,54 @@ def test_course_catalog_access_num_queries(self, user_attr_name, action, course_
         course_overview = CourseOverview.get_from_id(course.id)
         with self.assertNumQueries(num_queries, table_blacklist=QUERY_COUNT_TABLE_BLACKLIST):
             bool(access.has_access(user, action, course_overview, course_key=course.id))
+
+
+class CCXOverviewTestCase(SharedModuleStoreTestCase):
+    """
+    Tests for ccx course overviews
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def test_ccx_visibility(self):
+        """
+        Test to ensure that the ccx courses are hidden in course catalog.
+        """
+
+        course = CourseFactory.create(enable_ccx=True)
+        CourseOverview.load_from_module_store(course.id)
+
+        coach = AdminFactory.create()
+        ccx = CustomCourseForEdX(
+            course_id=course.id,
+            coach=coach,
+            display_name="Test CCX"
+        )
+        ccx.save()
+
+        student_password = "foobar"
+        student = UserFactory.create(username="test", password=student_password, is_staff=False)
+
+        ccx_overview = CourseOverview.get_from_id(ccx.locator)
+
+        # base course is still visible
```

Comment:
You can condense this test code substantially by using [`ddt`](https://ddt.readthedocs.io/en/latest/example.html), as demonstrated by the [AccessTestCase](https://github.com/edx/edx-platform/blob/65a8d6df64bece05cd79be3760e06f723ec9849c/lms/djangoapps/courseware/tests/test_access.py#L162-L163).

* Move your course and user creation into a `setUp` method ([like this](https://github.com/edx/edx-platform/blob/65a8d6df64bece05cd79be3760e06f723ec9849c/lms/djangoapps/courseware/tests/test_access.py#L80)).
* Add `staff` user too.
* Use `@ddt.data` and `@ddt.unpack` to test for all the various permutations of user and permission across these three user types and the two courses. e.g.,
```python
@ddt.data (
('student', 'see_in_catalog', True),
('student', 'see_exists', True),
('student', 'enroll', True),
('staff', 'see_in_catalog', True),
...
)
@ddt.unpack
def test_ccx_base_course_access(self, username, permission, has_permission):
    user = self.users[username]
    self.assertEqual(access.has_access(user, permission, course), has_permission)

@ddt.data (
('student', 'see_in_catalog', False),
..
)
@ddt.unpack
def test_ccx_child_course_access(self, username, permission, has_permission):
    user = self.users[username]
    self.assertEqual(access.has_access(user, permission, ccx_overview), has_permission)
```